yices2.dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM hdlc/build:base AS build
```

Comment:
FTR, `hdlc/build:build` includes `clang` and `make` on top of `hdlc/build:base`. Hence, it might be suitable to use that image instead, and avoid installing `g++` and `make` here.

However, additional configuration options might be required for the build scripts to use clang. Hence, this does not need to be done in this PR.

Reply:
In the `z3` dockerfile, same tools are installed.

Reply:
https://github.com/hdl/containers/commit/27b46296b47e25bad5f2d219713fb8e55645b47d